azurerm/resource_arm_storage_account.go

```diff
@@ -1406,13 +1406,14 @@ func expandQueuePropertiesCors(input []interface{}) *queues.Cors {
 }
 
 func flattenStorageAccountNetworkRules(input *storage.NetworkRuleSet) []interface{} {
-	if len(*input.IPRules) == 0 && len(*input.VirtualNetworkRules) == 0 {
-		return []interface{}{}
-	}
 	networkRules := make(map[string]interface{})
 
-	networkRules["ip_rules"] = schema.NewSet(schema.HashString, flattenStorageAccountIPRules(input.IPRules))
-	networkRules["virtual_network_subnet_ids"] = schema.NewSet(schema.HashString, flattenStorageAccountVirtualNetworks(input.VirtualNetworkRules))
+	if len(*input.IPRules) > 0 {
+		networkRules["ip_rules"] = schema.NewSet(schema.HashString, flattenStorageAccountIPRules(input.IPRules))
+	}
+	if len(*input.VirtualNetworkRules) > 0 {
+		networkRules["virtual_network_subnet_ids"] = schema.NewSet(schema.HashString, flattenStorageAccountVirtualNetworks(input.VirtualNetworkRules))
+	}
```

Comment:
can we remove this `if` check here? the flatten function will return an empty array when this is nil - which'll ensure we track diff's here:

```suggestion
	networkRules["virtual_network_subnet_ids"] = schema.NewSet(schema.HashString, flattenStorageAccountVirtualNetworks(input.VirtualNetworkRules))
```

Reply:
@tombuildsstuff now that i am removing the checks completely, the current test "TestAccAzureRMStorageAccount_networkRules" is actually good enough, i think. 

https://github.com/terraform-providers/terraform-provider-azurerm/blob/0fd80ac109826ceb3c2cd6ca791b5943e45ec0c4/azurerm/resource_arm_storage_account_test.go#L636

 previously, it didn't test the conditional branch and now that i'm removing the branches completely.

what do you think?